crates/evm/evm/src/executors/corpus.rs

```diff
@@ -525,64 +643,49 @@ impl CorpusManager {
             return self.new_tx(test_runner);
         }
 
-        // Continue with the next call initial sequence
+        // Continue with the next call initial sequence.
         Ok(sequence[depth].clone())
     }
 
-    /// Generates single call from corpus strategy.
-    pub fn new_tx(&mut self, test_runner: &mut TestRunner) -> eyre::Result<BasicTxDetails> {
-        Ok(self
-            .tx_generator
-            .new_tree(test_runner)
-            .map_err(|_| eyre!("Could not generate case"))?
-            .current())
-    }
-
-    /// Returns campaign failed replays.
-    pub fn failed_replays(self) -> usize {
-        self.failed_replays
-    }
-
-    /// Collects coverage from call result and updates metrics.
-    pub fn merge_edge_coverage(&mut self, call_result: &mut RawCallResult) -> bool {
-        if !self.config.collect_edge_coverage() {
-            return false;
-        }
-
-        let (new_coverage, is_edge) = call_result.merge_edge_coverage(&mut self.history_map);
-        if new_coverage {
-            self.metrics.update_seen(is_edge);
-        }
-        new_coverage
-    }
-
     /// Flush the oldest corpus mutated more than configured max mutations unless they are
     /// favored.
-    fn evict_oldest_corpus(&mut self) -> eyre::Result<()> {
+    fn evict_oldest_corpus(&mut self) -> Result<()> {
         if self.in_memory_corpus.len() > self.config.corpus_min_size.max(1)
             && let Some(index) = self.in_memory_corpus.iter().position(|corpus| {
                 corpus.total_mutations > self.config.corpus_min_mutations && !corpus.is_favored
             })
         {
-            let corpus = self.in_memory_corpus.get(index).unwrap();
-
-            let uuid = corpus.uuid;
-            debug!(target: "corpus", "evict corpus {uuid}");
+            let corpus = &self.in_memory_corpus[index];
 
+            // TODO(dani): metadata?
+            /*
             // Flush to disk the seed metadata at the time of eviction.
+            let uuid = corpus.uuid;
```

Comment:
let's then totally get rid of this, I think @0xalpharush mentioned not strictly required